doc/sfp-cmis/Interface-Link-bring-up-sequence.md

```diff
@@ -184,18 +185,246 @@ if transceiver is not present:
+      - For transceivers which do not support media settings, MEDIA_SETTINGS_SYNC_STATUS will stay with value MEDIA_SETTINGS_DEFAULT
+
+  Following table describes the various values for MEDIA_SETTINGS_SYNC_STATUS  
+
+|          Value          |                Modifier thread and event               |                                  Consumer thread and purpose                                 |
+|:-----------------------:|:------------------------------------------------------:|:--------------------------------------------------------------------------------------------:|
+| MEDIA_SETTINGS_DEFAULT  | XCVRD main thread during cold   start of xcvrd         | XCVRD main thread   during boot-up for deciding to notify media settings                     |
+|                         | SfpStateUpdateTask during   transceiver removal        |                                                                                              |
+| MEDIA_SETTINGS_NOTIFIED | SfpStateUpdateTask while   updating the media settings | Not being used currently                                                                     |
+| MEDIA_SETTINGS_DONE     | Orchagent after applying the SI   settings             | CmisManagerTask for proceeding   to CMIS_STATE_DP_DEINIT from CMIS_STATE_MEDIA_SETTINGS_WAIT |
+
+2. SfpStateUpdateTask thread will notify the media settings to OA based on the value of PORT_TABLE:\<port\>.MEDIA_SETTINGS_SYNC_STATUS  
+If PORT_TABLE:\<port\>.MEDIA_SETTINGS_SYNC_STATUS != MEDIA_SETTINGS_DONE, notify media settings will be invoked and will be set to MEDIA_SETTINGS_NOTIFIED for a port supporting media settings.
+3. The OA upon receiving media settings will
+	- Disable port admin status
+	- Apply SI settings
+	- PORT_TABLE:\<port\>.MEDIA_SETTINGS_SYNC_STATUS = MEDIA_SETTINGS_DONE
+4. In the CMIS_STATE_INSERTED state, if 'admin_status' is up and 'host_tx_ready' is true, CmisManagerTask thread will check if
+	- the port supports media settings (will be checked using g_dict and finding valid SI values) and
+	- MEDIA_SETTINGS_SYNC_STATUS != MEDIA_SETTINGS_DONE  
+If all the above conditions are true, CMIS SM transitions to CMIS_STATE_MEDIA_SETTINGS_WAIT state.  
+If port doesn't require media settings to be applied, CMIS SM will proceed with normal code flow (transitions to CMIS_STATE_DP_DEINIT)  
+Overall, no functionality change related to CMIS SM transitions is intended for ports not supporting media settings
+5. CMIS_STATE_MEDIA_SETTINGS_WAIT state will wait for MEDIA_SETTINGS_DONE and upon reaching to MEDIA_SETTINGS_DONE, CMIS SM will transition to CMIS_STATE_DP_DEINIT.  
+There will be a timeout of 5s for every retry
+6. The CmisManagerTask thread will set “CMIS_REINIT_REQUIRED" to false after CMIS SM reaches to a steady state (CMIS_STATE_UNKNOWN, CMIS_STATE_FAILED, CMIS_STATE_READY and CMIS_STATE_REMOVED) for the corresponding port
```

Comment:
Shouldn't CMIS_REINIT_REQUIRED be set to True for all states other than CMIS_STATE_READY, to allow CmisManagerTask to reinit the module if xcvrd crashes?

Reply:
I don't think so since CMIS_REINIT_REQUIRED is created mainly to force CMIS re-initialization in case of xcvrd/OA crash. Hence, if a port reaches to any of the steady states above (apart from CMIS_STATE_READY), it means that the port has genuinely failed initialization (CMIS_STATE_FAILED) or it doesn't require forced re-initialization (CMIS_STATE_REMOVED).
Any subsequent triggers for CMIS initialization would still be addressed via https://github.com/sonic-net/sonic-platform-daemons/blob/76baca3d71542dbeeeeac6bea45b762abab7166f/sonic-xcvrd/xcvrd/xcvrd.py#L1557.
This would still ensure that CMIS initialization gets triggered.